src/states/node.rs

```diff
@@ -1248,51 +1284,30 @@ impl Node {
         self.send_routing_message(response_msg)
     }
 
-    // Received by Y; From A -> Y, or from any node to one of its bucket addresses.
-    fn handle_get_close_group_request(&mut self,
-                                      src: Authority,
-                                      dst_name: XorName,
-                                      message_id: MessageId)
-                                      -> Result<(), RoutingError> {
-        // If msg is from ourselves, ignore it.
-        if src.name() == self.name() {
+    fn handle_section_update(&mut self,
+                             prefix: Prefix<XorName>,
+                             members: Vec<PublicId>)
+                             -> Result<(), RoutingError> {
+        // Filter list of members to just those we don't know about:
+        let members = if let Some(section) = self.peer_mgr.routing_table().section_ref(&prefix) {
+            let f = |id: &PublicId| !(section.is_member(id.name()) || section.is_needed(id.name()));
+            members.into_iter().filter(f).collect_vec()
+        } else {
+            warn!("{:?} Section update received from unknown neighbour {:?}", self, prefix);
             return Ok(());
-        }
-
-        let public_ids = match self.peer_mgr.routing_table().close_names(&dst_name) {
-            Some(close_group) => self.peer_mgr.get_pub_ids(&close_group),
-            None => return Err(RoutingError::InvalidDestination),
-        };
-
-        trace!("{:?} Sending GetCloseGroup response with {:?} to {:?}.",
-               self,
-               public_ids.iter().map(PublicId::name).collect_vec(),
-               src);
-        let response_content = MessageContent::GetCloseGroupResponse {
-            close_group_ids: public_ids.into_iter().collect(),
-            message_id: message_id,
         };
 
-        let response_msg = RoutingMessage {
-            src: Authority::ManagedNode(*self.name()),
-            dst: src,
-            content: response_content,
-        };
-
-        self.send_routing_message(response_msg)
-    }
-
-    fn handle_get_close_group_response(&mut self,
-                                       close_group_ids: Vec<PublicId>,
-                                       dst: Authority)
-                                       -> Result<(), RoutingError> {
-        for close_node_id in close_group_ids {
-            if self.peer_mgr.routing_table().need_to_add(close_node_id.name()).is_ok() {
-                debug!("{:?} Sending connection info to {:?} on GetCloseGroup response.",
-                       self,
-                       close_node_id);
-                let ci_dst = Authority::ManagedNode(*close_node_id.name());
-                self.send_connection_info(close_node_id, dst, ci_dst)?;
+        let own_name = *self.name();
+        for pub_id in members {
+            let name: XorName = *pub_id.name();
```

Comment:
Is the type annotation necessary here? Not exactly a problem, but looks a bit out-of-place.